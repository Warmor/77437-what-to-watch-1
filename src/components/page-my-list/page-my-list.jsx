import React from 'react';
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {Operation as OperationData} from "~/reducer/data/data";
import {getFavorite} from "~/reducer/favorite/selectors";
import Header from "~/components/header/header";
import Footer from "~/components/footer/footer";
import MoviesList from "~/components/movies-list/movies-list";

class PageMyList extends React.PureComponent {
  componentDidMount() {
    this.props.loadFavorite();
  }
  render() {
    return <>
      <div className="user-page">
        <Header headerTitle={`My list`} withAuth/>
        <MoviesList films={this.props.favorite}/>
        <Footer/>
      </div>
    </>;
  }
}

PageMyList.propTypes = {
  favorite: PropTypes.arrayOf(PropTypes.object).isRequired,
  loadFavorite: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) => ({
  ...ownProps,
  favorite: getFavorite(state)
});
const mapDispatchToProps = (dispatch) => ({
  loadFavorite: () => dispatch(OperationData.loadFavorite())
});

export {PageMyList};

export default connect(mapStateToProps, mapDispatchToProps)(PageMyList);
