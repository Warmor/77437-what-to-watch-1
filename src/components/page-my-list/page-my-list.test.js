import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallowToJson} from 'enzyme-to-json';
Enzyme.configure({adapter: new Adapter()});

import {PageMyList} from "~/components/page-my-list/page-my-list";

describe(`<PageMyList/>`, () => {
  it(`PageMyList renders correctly`, () => {
    const mockProps = {
      favorite: [{
        id: 1,
        title: `Fantastic Beasts`,
        genreId: 1,
        image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
        video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
      }],
      loadFavorite: jest.fn()
    };
    const tree = shallow(<PageMyList
      favorite={mockProps.favorite}
      loadFavorite={mockProps.loadFavorite}/>);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
