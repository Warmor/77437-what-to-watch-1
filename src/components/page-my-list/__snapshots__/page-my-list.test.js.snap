// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageMyList/> PageMyList renders correctly 1`] = `
<Fragment>
  <div
    className="user-page"
  >
    <ConnectFunction
      headerTitle="My list"
      withAuth={true}
    />
    <MoviesList
      films={
        Array [
          Object {
            "genreId": 1,
            "id": 1,
            "image": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
            "title": "Fantastic Beasts",
            "video": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          },
        ]
      }
    />
    <Footer />
  </div>
</Fragment>
`;
